arcgis-ios-sdk-samples/Layers/Change sublayer renderer/ChangeSublayerRendererVC.swift

```diff
@@ -48,33 +50,49 @@ class ChangeSublayerRendererVC: UIViewController {
         
         //load map image layer to access sublayers
         self.mapImageLayer.load { [weak self] (error: Error?) in
-            
-            guard let weakSelf = self else {
-                return
+            guard let self = self else { return }
+            if let error = error {
+                print("Error loading map image layer: \(error)")
+            } else {
+                self.mapImageLayerDidLoad()
             }
-            
-            //get the counties sublayer
-            if let sublayer = weakSelf.mapImageLayer.mapImageSublayers[2] as? AGSArcGISMapImageSublayer {
-                
-                //load the sublayer to get the original renderer
-                sublayer.load(completion: { (error) in
-                    if error == nil {
-                        weakSelf.originalRenderer = sublayer.renderer
-                        
-                        //enable bar button items
-                        self?.applyRendererBarButtonItem.isEnabled = true
-                        self?.resetBarButtonItem.isEnabled = true
-                    }
-                })
+        }
+    }
+    
+    /// Called in response to the map image layer loading successfully.
+    func mapImageLayerDidLoad() {
+        //get the counties sublayer
+        let mapImageSublayers = mapImageLayer.mapImageSublayers
+        guard mapImageSublayers.count >= 3,
+            let sublayer = mapImageSublayers[2] as? AGSArcGISMapImageSublayer else {
+                return
+        }
+        countiesSublayer = sublayer
+        //load the sublayer to get the original renderer
+        sublayer.load { [weak self] (error) in
+            guard let self = self else { return }
+            if let error = error {
+                print("Error loading map image sublayer: \(error)")
```

Comment:
Should show error to user?

Reply:
That's a good question. We don't use `SVProgressHUD` to show errors anymore. We do use `UIAlertController` to show some errors, but usually only errors that the user can do something about. In the event of a load error, the user can't really do anything about it (other than close and reopen the sample). And since a load error is unlikely to occur, it is simpler to just log those errors rather than try to show them to the user.

Reply:
ok, thanks for clarification.